File: seminars/Sem07_OOP_Desing_and_SOLID_part_2/OnlineTasks/FromJavaToPython/Model/Tea.py
from .HotDrink import HotDrink


class Tea(HotDrink):

    def __init__(self, my_id, temperature, cost, is_ready_to_serve, is_green, region):
        super().__init__(my_id, temperature, cost, is_ready_to_serve)
        self.is_green = is_green
        self.region = region

    def get_id(self):
        return self.id

    def __str__(self) -> str:
        return (f'Tea [id={self.id}'
                f', temperature={self.temperature}'
                f', cost={self.cost}'
                f', is_ready_to_serve={self.is_ready_to_serve}'
                f', is_green={self.is_green}'
                f', region={self.region}]')

File: seminars/Sem07_OOP_Desing_and_SOLID_part_2/OnlineTasks/src/Application/Entities/Tea.java
package Application.Entities;

import Application.Abstracts.HotDrink;

public class Tea extends HotDrink {

    private boolean isGreen;
    private String region;

    public Tea(int id, int temperature, float coast, boolean isGreen, String region) {
        super(id, temperature, coast);
        setGreen(isGreen);
        setRegion(region);
    }

    public boolean isGreen() {
        return isGreen;
    }

    public void setGreen(boolean green) {
        isGreen = green;
    }

    public String getRegion() {
        return region;
    }

    public void setRegion(String region) {
        this.region = region;
    }

    @Override
    public String toString() {
        return "Tea{" +
                "id=" + getId() +
                ", coast=" + getCoast() +
                ", temperature=" + getTemperature() +
                ", isGreen=" + isGreen() +
                ", region='" + getRegion() + '\'' +
                '}';
    }
}
